c/extras/py_ftrl.cc

```diff
@@ -23,127 +23,220 @@
-
-  gs.add<&Ftrl::get_a, &Ftrl::set_a>("a", "`alpha` in per-coordinate learning rate formula.\n");
-  gs.add<&Ftrl::get_b, &Ftrl::set_b>("b", "`beta` in per-coordinate learning rate formula.\n");
-  gs.add<&Ftrl::get_l1, &Ftrl::set_l1>("l1", "L1 regularization parameter.\n");
-  gs.add<&Ftrl::get_l2, &Ftrl::set_l2>("l2", "L2 regularization parameter.\n");
-  gs.add<&Ftrl::get_d, &Ftrl::set_d>("d", "Number of bins to be used after the hashing trick.\n");
-  gs.add<&Ftrl::get_n_epochs, &Ftrl::set_n_epochs>("n_epochs", "Number of epochs to train for.\n");
-  gs.add<&Ftrl::get_inter, &Ftrl::set_inter>("inter", "If feature interactions to be used or not.\n");
-  gs.add<&Ftrl::get_hash_type, &Ftrl::set_hash_type>("hash_type", "Hashing method to use for strings.\n"
+    "must be `FLOAT64`"
+  );
+
+  gs.add<&PyFtrl::get_params, &PyFtrl::set_params>(
+    "params",
+    "FTRL model parameters"
+  );
+
+  gs.add<&PyFtrl::get_default_params>(
+    "default_params",
+    "FTRL model default parameters"
+  );
+
+  gs.add<&PyFtrl::get_colnames_hashes>(
+    "colnames_hashes",
+    "Column name hashes.\n"
+  );
+
+  gs.add<&PyFtrl::get_alpha, &PyFtrl::set_alpha>(
+    params_fields_info[0].first,
+    params_fields_info[0].second
+  );
+
+  gs.add<&PyFtrl::get_beta, &PyFtrl::set_beta>(
+    params_fields_info[1].first,
+    params_fields_info[1].second
+  );
+
+  gs.add<&PyFtrl::get_lambda1, &PyFtrl::set_lambda1>(
+    params_fields_info[2].first,
+    params_fields_info[2].second
+  );
+
+  gs.add<&PyFtrl::get_lambda2, &PyFtrl::set_lambda2>(
+    params_fields_info[3].first,
+    params_fields_info[3].second
+  );
+
+  gs.add<&PyFtrl::get_d, &PyFtrl::set_d>(
+    params_fields_info[4].first,
+    params_fields_info[4].second
+  );
+
+  gs.add<&PyFtrl::get_n_epochs, &PyFtrl::set_n_epochs>(
+    params_fields_info[5].first,
+    params_fields_info[5].second
+  );
+
+
+  gs.add<&PyFtrl::get_inter, &PyFtrl::set_inter>(
+    params_fields_info[6].first,
+    params_fields_info[6].second
+  );
+
+  gs.add<&PyFtrl::get_hash_type, &PyFtrl::set_hash_type>(
+    "hash_type",
+    "Hashing method to use for strings\n"
     "`0` - std::hash;\n"
     "`1` - Murmur2;\n"
-    "`2` - Murmur3.\n");
-  gs.add<&Ftrl::get_seed, &Ftrl::set_seed>("seed", "Seed to be used for Murmur hash functions.\n");
-
-  mm.add<&Ftrl::fit, args_fit>();
-  mm.add<&Ftrl::predict, args_predict>();
-  mm.add<&Ftrl::reset, args_reset>();
+    "`2` - Murmur3.\n"
+  );
+
+  gs.add<&PyFtrl::get_seed, &PyFtrl::set_seed>(
+    "seed",
+    "Seed to be used for Murmur hash functions\n"
+  );
+
+  mm.add<&PyFtrl::fit, args_fit>();
+  mm.add<&PyFtrl::predict, args_predict>();
+  mm.add<&PyFtrl::reset_model, args_reset_model>();
+  mm.add<&PyFtrl::reset_params, args_reset_params>();
 }
 
 
-PKArgs Ftrl::Type::args_fit(1, 0, 0, false, false, {"frame"}, "fit",
+PKArgs PyFtrl::Type::args_fit(1, 0, 0, false, false, {"frame"}, "fit",
 R"(fit(self, frame)
```

Comment:
The standard scikit signature is `fit(X, y)` (note that `X` is capitalized). The target column `y` is provided by the user separately

Reply:
Do you know if this signature is also something standard in DAI for other algos?

Reply:
it's used in DAI's transformers, and also in `h2o`s models

Reply:
Gotcha, then let me make it closer to the standard.